File: src/main/java/ru/qwonix/rentucha/api/entity/Apartment.java
package ru.qwonix.rentucha.api.entity;

import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Data
public class Apartment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, length = 2000)
    private String description;

    @Column(nullable = false)
    private String countryName;

    @Column(nullable = false)
    private String cityName;

    @Column(nullable = false)
    private Double pricePerNight;

    @Column(nullable = false)
    private Double maxPersonCount;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    @Column
    private String mainImageUrl;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime creationTimestamp;
}

File: app/src/main/java/ru/qwonix/android/rentucha/entity/Apartment.kt
package ru.qwonix.android.rentucha.entity

import java.time.LocalDateTime
import java.time.temporal.ChronoUnit


data class Apartment(
//    var id: Long,
//    var name: String,
//    var description: String,
    var countryName: String,
    var cityName: String,
    var pricePerNight: Double,
    var mainImageUrl: String,
    var maxPersonCount: Int,
    var latitude: Double,
    var longitude: Double,

    var creationTimestamp: LocalDateTime

//    var images: List<String>,

) {
    fun getAddedWeeksAgo(): Long {
        return ChronoUnit.WEEKS.between(
            creationTimestamp,
            LocalDateTime.now()
        )
    }
}